frontend/src/v5/ui/components/viewer/drawingViewer/viewerLayer2D/camera/camera.component.tsx

```diff
@@ -0,0 +1,121 @@
+/**
+ *  Copyright (C) 2024 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { DrawingsHooksSelectors } from '@/v5/services/selectorsHooks';
+import { useSearchParam } from '@/v5/ui/routes/useSearchParam';
+import { useRef, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { Viewer as ViewerService } from '@/v4/services/viewer/viewer';
+import { Vector3 } from 'three';
+import { useModelLoading, useViewpointSubscription } from './viewer.hooks';
+import { CameraIcon } from './camera.styles';
+import { ViewerParams } from '@/v5/ui/routes/routes.constants';
+
+
+
+const haloAngleCutoff = Math.PI / 4;
+
+export const Camera = ({ scale }) => {
+	const viewpoint = useRef(null);
+	
+	const [drawingId] = useSearchParam('drawingId');
+	const { containerOrFederation } = useParams<ViewerParams>();
+	const [position, setPosition] = useState({ x:0, y:0 });
+	const [angle, setAngle] = useState(0);
+	const [haloVisibility, setHaloVisibility] = useState(1);
+	const transform2DTo3D = DrawingsHooksSelectors.selectTransform2Dto3D(drawingId, containerOrFederation);
+	const transform3DTo2D = DrawingsHooksSelectors.selectTransform3Dto2D(drawingId, containerOrFederation);
+
+	const modelLoading = useModelLoading();
+
+	useViewpointSubscription((v) => {
+		if (!transform3DTo2D) return;
+
+		const p =  transform3DTo2D(v.position);
```

Comment:
Should the camera (which refers to the previous calibration) be hidden when calibrating?
![image](https://github.com/user-attachments/assets/33d4608e-9f26-4290-8190-406bea065271)